### ISSUE TITLE
when node docker restarted the before created pod's wait container exit unexcept

### ISSUE THREAD

**Issue by huangjiasingle** [argoproj/argo-workflows#9129]
## Checklist

 

* [ ] Double-checked my configuration.
* [ ] Tested using the latest version.
* [ ] Used the Emissary executor.

## Summary

What happened/what you expected to happen?
the node docker restart the wait container exit not except
What version are you running?
v3.3.8
## Diagnostics

the wait container mount node's /var/run/docker.sock in container's /var/run/docker.sock, when docker restart the node's docker.sock was change, bug in the wait container's  docker.sock is old, so wait container wait command return: `docker wait c2c6dd18a852585bb03f6f47084f962179067ea8dab14cc102b5663631392142` failed: unexpected EOF`.   

```yaml

```

```bash
# Logs from the workflow controller:
kubectl logs -n argo deploy/workflow-controller | grep ${workflow} 

# If the workflow's pods have not been created, you can skip the rest of the diagnostics.

# The workflow's pods that are problematic:
kubectl get pod -o yaml -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded

# Logs from in your workflow's wait container, something like:

```
time="2022-07-08T17:40:42.994Z" level=info msg="Alloc=4823 TotalAlloc=79897 Sys=75857 NumGC=123 Goroutines=11"
time="2022-07-08T17:45:42.994Z" level=info msg="Alloc=4536 TotalAlloc=81347 Sys=75857 NumGC=126 Goroutines=11"
time="2022-07-08T17:50:42.994Z" level=info msg="Alloc=4776 TotalAlloc=82749 Sys=75857 NumGC=128 Goroutines=11"
time="2022-07-08T17:55:42.995Z" level=info msg="Alloc=4490 TotalAlloc=84200 Sys=75857 NumGC=131 Goroutines=11"
time="2022-07-08T17:55:44.215Z" level=error msg="`docker wait c2c6dd18a852585bb03f6f47084f962179067ea8dab14cc102b5663631392142` failed: unexpected EOF\n"
---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by terrytangyuan** [comment]
Please use the Emissary executor and feel free to open if you still have issues.

**Comment by huangjiasingle** [comment]
> @huangjiasingle are you using docker as the main container? can you provide workflow yaml and pod yaml?

yes,  but it doesn't matter, I have fixed the problem myself by modifying the argo docker exec  code

**Comment by huangjiasingle** [comment]
> Please use the Emissary executor and feel free to open if you still have issues.

l can't use emissary executor to do in prod env, because every day we has more than 1000000 prod data process task to do with argo workflow, I haven't tested it with emissary executor for a lot of tasks. first of all, I need to ensure the stability of production task processing.
